### PR TITLE
Add CompressionLevel and make v2 Kafka sink default

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-11339, DOC-10867, DOC-10830, DOC-10700

This PR:

- Updates the `kafka_sink_config` option with the `CompressionLevel` field in v24.3. 
- Removes the note for the cluster setting to enable the v2 Kafka sink, because this is the default in v24.3.
- Adds the cluster setting `changefeed.sink-io-workers` under Kafka for the default v2 sink.

## Rendered Preview

https://deploy-preview-19169--cockroachdb-docs.netlify.app/docs/v24.3/changefeed-sinks.html#kafka-sink-configuration

### CODE REVIEW

File: src/current/v24.3/changefeed-sinks.md [hunk 1]
```diff
@@ -117,14 +106,17 @@ Kafka has the following topic limitations:
 
 ### Kafka sink configuration
 
- The `kafka_sink_config` option allows configuration of a changefeed's message delivery, Kafka server version, and batching parameters.
+You can configure flushing, acknowledgments, compression, and concurrency behavior of changefeeds running to a Kafka sink with the following:
+
+- Set the [`changefeed.sink_io_workers` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-changefeed-sink-io-workers) to configure the number of concurrent workers used by changefeeds in the cluster when sending requests to a Kafka sink. When you set `changefeed.sink_io_workers`, it will not affect running changefeeds; [pause the changefeed]({% link {{ page.version.version }}/pause-job.md %}), set `changefeed.sink_io_workers`, and then [resume the changefeed]({% link {{ page.version.version }}/resume-job.md %}). Note that this cluster setting will also affect changefeeds running to [Google Cloud Pub/Sub](#google-cloud-pub-sub) sinks and [webhook sinks](#webhook-sink).
```
Comment (kathancox) [inline src/current/v24.3/changefeed-sinks.md:111]: Does it make sense to add this now that the v2 Kafka sink is the default? (This paragraph is included for Pub/Sub + Webhook too.)

File: src/current/v24.3/changefeed-sinks.md [hunk 2]
```diff
@@ -154,6 +146,7 @@ Field              | Type                | Description      | Default
...
+<span class="version-tag">New in v24.3:</span>`"CompressionLevel"` | [`INT`]({% link {{ page.version.version }}/int.md %}) | Sets the level of compression. This determines the level of compression ratio versus compression speed, i.e., how much the data size is reduced (better compression) and how quickly the compression process is completed. The compression protocols have the following ranges:<br>`GZIP`:<ul><li>`0` no compression</li><li>`1` to `9` best speed to best compression</li><li>`-1` default</li><li>`-2` [Huffman-only compression](https://en.wikipedia.org/wiki/Huffman_coding)</li></ul>`ZSTD`:<ul><li>`1` fastest</li><li>`2` default</li><li>`3` better compression</li><li>`4` best compression</li></ul>`LZ4`<ul><li>0 fast default</li><li>`512 * N` Level N, where N is between `1` and `9`. The higher the number, the better compression</li></ul>**Note:** If you have the `changefeed.new_kafka_sink.enabled` cluster setting disabled, `CompressionLevel` will not affect `LZ4` compression. `SNAPPY` does not support `CompressionLevel`. | `GZIP`: `-1`<br><br>`ZSTD`: `2`<br><br>`LZ4`: `0`
```
[the remark this answers is not in the document]
Comment (kathancox) [reply src/current/v24.3/changefeed-sinks.md]: @asg0451 I changed this up a bit, the table for all the information wasn't working so I added some subsections with the table acting more as a list. You'll have to excuse the diff noise of the other field subsections. The `LZ4` values, I haven't added those — I've instead stuck to the same format as `GZIP` using the `LZ4` levels, let me know what you think.

For the default `GZIP` value, I've left it as `-1`, but noted you can't manually set this as such — this is what I interpreted from your Slack message, so I may be wrong here.

I added the known limitation and tried to be clear about how this does/doesn't apply. 

PTAL! Here's the preview: https://deploy-preview-19169--cockroachdb-docs.netlify.app/docs/v24.3/changefeed-sinks.html#kafka-sink-configuration

File: src/current/_includes/v24.3/known-limitations/compression-level-kafka-config.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Changefeeds created in v24.3 of CockroachDB that emit to [Kafka]({% link {{ page.version.version }}/changefeed-sinks.md %}#kafka), or changefeeds created in earlier versions with the `changefeed.new_kafka_sink.enabled` cluster setting enabled, do not support negative compression level values in the [`kafka_sink_config = {... "CompressionLevel" = ...}`]({% link {{ page.version.version }}/changefeed-sinks.md %}#compressionlevel) option field. [#136492](https://github.com/cockroachdb/cockroach/issues/136492)
```
Comment (asg0451) [inline src/current/_includes/v24.3/known-limitations/compression-level-kafka-config.md:1]: can you call out that this is specifically for GZIP pls

File: src/current/v24.3/changefeed-sinks.md (hunk 1 above)
Comment (asg0451) [reply src/current/v24.3/changefeed-sinks.md:111]: please add the caveat that this only applies if running with the v2 kafka sink

File: src/current/v24.3/changefeed-sinks.md [hunk 4]
```diff
@@ -117,17 +106,36 @@ Kafka has the following topic limitations:
...
+<a name="kafka-compression"></a>`"Compression"` | [`STRING`]({% link {{ page.version.version }}/string.md %}) | Sets a compression protocol that the changefeed should use when emitting events. The possible values are: `"NONE"`, `"GZIP"`, `"SNAPPY"`, `"LZ4"`, `"ZSTD"`. | `"NONE"`
+<span class="version-tag">New in v24.3:</span>`"CompressionLevel"` | [`INT`]({% link {{ page.version.version }}/int.md %}) | Sets the level of compression. This determines the level of compression ratio versus compression speed, i.e., how much the data size is reduced (better compression) and how quickly the compression process is completed. For the compression protocol ranges, refer to [`CompressionLevel`](#compressionlevel).<br><br>**Note:** If you have the `changefeed.new_kafka_sink.enabled` cluster setting disabled, `CompressionLevel` will not affect `LZ4` compression. `SNAPPY` does not support `CompressionLevel`. | `GZIP`: `-1`<br><br>`ZSTD`: `2`<br><br>`LZ4`: `0`
```
Comment (asg0451) [inline src/current/v24.3/changefeed-sinks.md:130]: while accurate, i'm not sure how i feel about listing GZIP's default as -1 given that the user can't set that with the v2 sink. maybe there's a better way to explain the defaults than to list the magic values.

**Review by asg0451 — CHANGES_REQUESTED** [review]

File: src/current/v24.3/changefeed-sinks.md (hunk 1 above)
Comment (kathancox) [reply src/current/v24.3/changefeed-sinks.md:111]: We're not using the terminology `v2` Kafka sink, so I'll reference the cluster setting etc. here.

File: src/current/v24.3/changefeed-sinks.md (hunk 4 above)
Comment (kathancox) [reply src/current/v24.3/changefeed-sinks.md:130]: I think this is going to be pretty awkward given there are "default" compressions. Throughout the docs, we provide the default values for options/settings etc.

File: src/current/v24.3/changefeed-sinks.md (hunk 1 above)
Comment (asg0451) [reply src/current/v24.3/changefeed-sinks.md:111]: sounds good thanks. should i update the linked issue to use that terminology too?

File: src/current/v24.3/changefeed-sinks.md (hunk 4 above)
Comment (asg0451) [reply src/current/v24.3/changefeed-sinks.md:130]: yeah i know. kinda awkward either way i guess. up to you

File: src/current/v24.3/changefeed-sinks.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (asg0451) [reply src/current/v24.3/changefeed-sinks.md]: sorry for the confusion -- in the screenshot above the values you specify are the ones in grey, not the constant names. so for "Level1" you specify 1024, etc.

File: src/current/v24.3/changefeed-sinks.md (hunk 4 above)
Comment (asg0451) [reply src/current/v24.3/changefeed-sinks.md:130]: fair enough, your call

File: src/current/v24.3/changefeed-sinks.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kathancox) [reply src/current/v24.3/changefeed-sinks.md]: Ahhhh, I'm sorry, I did not grasp that. I will update.

File: src/current/v24.3/changefeed-sinks.md [hunk 5]
```diff
@@ -117,17 +106,41 @@ Kafka has the following topic limitations:
...
+You can configure flushing, acknowledgments, compression, and concurrency behavior of changefeeds running to a Kafka sink with the following:
+
+- Set the [`changefeed.sink_io_workers` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-changefeed-sink-io-workers) to configure the number of concurrent workers used by changefeeds in the cluster when sending requests to a Kafka sink. When you set `changefeed.sink_io_workers`, it will not affect running changefeeds; [pause the changefeed]({% link {{ page.version.version }}/pause-job.md %}), set `changefeed.sink_io_workers`, and then [resume the changefeed]({% link {{ page.version.version }}/resume-job.md %}). `changefeed.sink_io_workers` will also affect changefeeds running to [Google Cloud Pub/Sub](#google-cloud-pub-sub) sinks and [webhook sinks](#webhook-sink).
+
+    {{site.data.alerts.callout_info}}
+    `changefeed.sink_io_workers` only applies to Kafka sinks created in v24.2.1+, or if the `changefeed.new_kafka_sink.enabled` cluster setting has been enabled in CockroachDB clusters running v23.2.10+ and v24.1.4+. 
+    {{site.data.alerts.end}}
```
Comment (kathancox) [inline src/current/v24.3/changefeed-sinks.md:115]: Added this note above re how the concurrent worker cluster setting interacts with the newer Kafka sink. I _think_ I have the versioning and such correct here.

File: src/current/v24.3/changefeed-sinks.md [hunk 6]
```diff
@@ -117,17 +106,41 @@ Kafka has the following topic limitations:
...
+<a name="kafka-compression"></a>`"Compression"` | [`STRING`]({% link {{ page.version.version }}/string.md %}) | Sets a compression protocol that the changefeed should use when emitting events. The possible values are: `"NONE"`, `"GZIP"`, `"SNAPPY"`, `"LZ4"`, `"ZSTD"`. | `"NONE"`
+<span class="version-tag">New in v24.3:</span>`"CompressionLevel"` | [`INT`]({% link {{ page.version.version }}/int.md %}) | Sets the level of compression. This determines the level of compression ratio versus compression speed, i.e., how much the data size is reduced (better compression) and how quickly the compression process is completed. For the compression protocol ranges, refer to [`CompressionLevel`](#compressionlevel).<br><br>**Note:** If you have the `changefeed.new_kafka_sink.enabled` cluster setting disabled, `CompressionLevel` will not affect `LZ4` compression. `SNAPPY` does not support `CompressionLevel`. | Refer to [`CompressionLevel`](#compressionlevel)
```
Comment (kathancox) [inline src/current/v24.3/changefeed-sinks.md:135]: For the Default column here, I took out the values and referred readers to the section with the fuller explanations (including the description re the known limitation for GZIP). I hope that's a good compromise, i.e, removing the default values without context and listing the defaults only in the section with the context.

File: src/current/v24.3/changefeed-sinks.md [hunk 7]
```diff
@@ -136,24 +149,55 @@ Implement a Kafka resource usage limit per changefeed by setting a client ID and
...
+
+- `GZIP`:
+    - `0`: No compression
+    - `1` to `9`: From fastest compression to best compression
+    {% comment %}
+    These values are not available yet per KL #136492
+    - `-1`: Default compression
+    - `-2`: [Huffman-only compression](https://en.wikipedia.org/wiki/Huffman_coding)
+    - `-3`: Stateless compression
+    {% endcomment %}
```
Comment (kathancox) [inline src/current/v24.3/changefeed-sinks.md:168]: I've commented this out, but can fully remove.

File: src/current/v24.3/changefeed-sinks.md [hunk 8]
```diff
@@ -136,24 +149,55 @@ Implement a Kafka resource usage limit per changefeed by setting a client ID and
...
+    - `0`: Fastest compression (Default)
+    - `512`
+    - `1024`
+    - `2048`
+    - `4096`
+    - `8192`
+    - `16384`
+    - `32768`
+    - `65536`
+    - `131072`: Best compression
```
Comment (kathancox) [inline src/current/v24.3/changefeed-sinks.md:185]: Hope I have now understood this!

**Review by asg0451 — APPROVED** [review]

**Review by rmloveland — APPROVED** [review]
LGTM!

**Pull request merged**